src/components/BlogList.jsx

```diff
@@ -1,22 +1,34 @@
 import BlogPost from "./BlogPost";
 import Pagination from "./Pagination";
-import React from "react";
+import React, { useState } from "react";
 import blogs from "../data/blogs.json";
 
 const PAGE_SIZES = [15, 25, 50, 100];
 
 function BlogList() {
-  const currentPaginationData = blogs.posts.slice(0, 15);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [rowsPerPage, setRowsPerPage] = useState(15);
+  const currentPaginationData = blogs.posts.slice(
```

Comment:
You can avoid recalculating (slicing) on every render by placing the blog posts data in a state.